packages/template-retail-react-app/package.json

```diff
@@ -55,7 +55,13 @@
         "react-router-dom": "^5.3.4"
     },
     "peerDependencies": {
-        "@chakra-ui/system": "^1.12.1"
+        "@chakra-ui/system": "^1.12.1",
+        "@tanstack/react-query": "^4.28.0"
+    },
+    "peerDependenciesMeta": {
+        "@tanstack/react-query": {
+            "optional": true
+        }
```

Comment:
This is the material difference in this PR, compared to #1492 and #1494. The other two address the issue by modifying the webpack config (not changed in this PR).